src/tests/test_simulation.py

```diff
@@ -43,40 +44,21 @@ def test_error_end_before_first_step(tmp_path):
                          population_size=100)
 
 
-def test_can_track(tmp_path):
-    """Check that we can get outputs from tracked attributes."""
-    size = 100
-    start = date.today()
-    step = timedelta(days=30)
-    end = start + step
-    config = SimulationConfig(size, start, end, tmp_path, step, ['num_alive'])
-    pop, results = run_simulation(config)
-    assert 'num_alive' in results
-
-
-def test_error_tracking_nonexistent(tmp_path):
-    """Check that we error if trying to track attributes that don't exist."""
-    size = 100
-    start = date.today()
-    step = timedelta(days=30)
-    end = start + step
-    config = SimulationConfig(size, start, end, tmp_path, step, ['NOT_AN_ATTR'])
-    with pytest.raises(SimulationException,
-                       match="Unrecognised tracked attribute"):
-        run_simulation(config)
-
-
 def test_death_occurs(tmp_path):
     """Check that the number of people alive always decreases."""
     # FIXME This will not necessarily be true once we add in births
     size = 10000
     start = date.today()
     step = timedelta(days=90)
     end = start + 200 * step
-    config = SimulationConfig(size, start, end, tmp_path, step, ['num_alive'])
-    pop, results = run_simulation(config)
+    config = SimulationConfig(size, start, end, tmp_path, step)
+    simulation_handler = SimulationHandler(config)
+    simulation_handler.run()
+    pop = simulation_handler.population
     # Check that the number alive never grows... (some steps may have 0 deaths)
-    assert all(results.num_alive.diff()[1:] <= 0)
+    # assert all(results.num_alive.diff()[1:] <= 0)
+    num_not_dead = sum(pop.data.date_of_death.isnull())
+    assert num_not_dead < len(pop.data)
```

Comment:
Can we check this from the SimulationOutput instead? (If we don't store the number of people alive there, we should! though that can be a different PR)

Reply:
I think we can add in something that checks the simulation output too, but I'm inclined to also check that the population object is itself updating correctly as that's the thing that's going to be accessed by other parts of the simulation that are affected by this.